CMakeLists.txt

```diff
@@ -125,27 +125,32 @@ endif()
 #----------------------------------------------------------
 # CUDA support
 #----------------------------------------------------------
-option(ARB_WITH_CUDA "use CUDA for GPU offload" OFF)
-if(ARB_WITH_CUDA)
-    find_package(CUDA REQUIRED)
-
-    # Turn off annoying and incorrect warnings generated in the JSON file.
-    # We also work around the same issue with the intel compiler.
-    set(CUDA_NVCC_FLAGS ${CUDA_NVCC_FLAGS};-Xcudafe \"--diag_suppress=not_used_in_template_function_params\";-Xcudafe \"--diag_suppress=cast_to_qualified_type\")
+set(ARB_GPU_TARGET "none" CACHE STRING "The target GPU architecture: one of {none,K20,K80,P100}")
```

Comment:
I assume you are proposing that the user would specify `sm_35`, `sm_60` etc.

I see your point, but on the other hand, using the name of the target GPU is more user friendly than asking users to specifically provide the `sm_X` architecture (I had to spend a few minutes googling to find out exactly what to use with K80). People know which GPU they have, not which architecture they are targetting.

The other reason I went for `ARB_GPU_TARGET` and GPU makes is because it isn't NVIDIA specific. When/if we support say AMD GPUs, the user names the AMD GPU, and the correct flags etc are chosen.

At the end of the day, the only `35` is the K20 GPUs, `37` is only for K80, and `60` is only for P100.

Reply:
Could we maybe just change the name to 'ARB_GPU_MODEL'?

Reply:
Yes.